### DISCUSSION TITLE
Hide content from presentation

### DISCUSSION THREAD

**Discussion by GregHilston** [maaslalani/slides#157] [Q&A] [answered]
Is there a way to write a, say comment, in a slide that does **not** render in the live presentation? 

I've seen this feature in alternative tools as a in-line note used when making slides, or even as a "speaker note" that is rendered separately for the speaker.

**Comment by marcotrosi** [comment]
https://github.com/maaslalani/slides/issues/97

had similar thoughts, see linked issue

**Reply by GregHilston** [reply]
Hey @marcotrosi I think this is slightly different. I'm not specifically talking about hiding/unhiding a specific slide, rather a subset of content in a slide. For example., imagine I had a slide that looked like this:

```
---
"The quick brown fox jumps over the lazy dog"

<some syntax to indicate that this shouldn't be rendered>
What's interesting about this sentence is that it uses every letter in the English alphabet exactly once
</some syntax to indicate that this shouldn't be rendered>
---
```

The first part of this pitch is that I'd be nice to put notes in the source code, just for the writer. The second part would be displaying this notes in something on a different process, so the speaker could reference these while presenting. 

Does that make more sense?

**Reply by marcotrosi** [reply]
yeah makes sense, and I noticed it's not the same. But still it could be, depending on the implementation, that the developer decides to have common parts of the implementation and syntax.
In my case it was only about hiding whole slides, e.g. when you have to shorten your presentation or a slide is not fully ready or whatever.

**Reply by GregHilston** [reply]
Understood. Would you like me to make a new github issue for this slightly different feature request? Or should I add this description to the existing one?

**Reply by marcotrosi** [reply]
> Understood. Would you like me to make a new github issue for this slightly different feature request? Or should I add this description to the existing one? 

Thanks for the offer. No actions required from your side. I have my old issue which got rejected for now. If it would ever get more urgent I would take care of it. But thanks, I very much appreciate your offer.

**Comment by maaslalani** [comment, marked as answer]
I will take a look at adding comments to slides, should not be too difficult to simply strip all comments from the slides and I think a reasonable feature suggestion 👍 

If you're looking for a quick workaround you can use [pre-processing](https://github.com/maaslalani/slides#pre-processing)

```markdown
# Test

~~~echo
Hidden Content
~~~

Visible Content
```

**Reply by GregHilston** [reply]
@maaslalani Ah that's true, I forgot about how echo's output of nothing will be written to stdout, and override the "Hidden Content" portion. Good recommendation!
